webapp/channels/src/components/user_settings/advanced/index.ts

```diff
@@ -34,22 +33,17 @@ function makeMapStateToProps(state: GlobalState, props: OwnProps) {
         const enableUserDeactivation = config.EnableUserDeactivation === 'true';
         const enableJoinLeaveMessage = config.EnableJoinLeaveMessageByDefault === 'true';
 
-        let getPreference = (prefCategory: string, prefName: string, defaultValue: string) => get(state, prefCategory, prefName, defaultValue);
-        if (props.adminMode && props.userPreferences) {
-            // This ties the function to the current value of userPreferences for the current execution of this function
-            const preferences = props.userPreferences;
-            getPreference = (prefCategory, prefName, defaultValue) => getFromPreferences(preferences, prefCategory, prefName, defaultValue);
-        }
+        const userPreferences = props.adminMode && props.userPreferences ? props.userPreferences : undefined;
```

Comment:
Do we even need to check `adminMode` here now? I wouldn't expect `userPreferences` to be passed unless `adminMode` was also true

Reply:
Just for clarifying intention.